tmt/base.py

```diff
@@ -2130,6 +2130,21 @@ def lint_unknown_keys(self) -> LinterReturn:
 
         yield LinterOutcome.PASS, 'correct keys are used'
 
+    def lint_empty_env_files(self) -> LinterReturn:
+        """ P001: env files are not empty """
+
+        env_files = self.node.get("environment-file") or []
+        if env_files:
+            for env_file in env_files:
+                env_file = Path(env_file).resolve()
+
+                if not env_file.exists() or not env_file.stat().st_size:
+                    yield LinterOutcome.FAIL, f'the file "{env_file}" does not exists or is empty'
+
+                    return
+
+        yield LinterOutcome.PASS, 'no empty environment files found'
```

Comment:
This sounds rather like a `SKIP` doesn't it? If there are no environment files there's nothing to check. What about handling this before the loop?

```python
if not env_files:
    yield LinterOutcome.SKIP, 'no environment files found'
    return
```

Reply:
Makes sense, it is more like a `SKIP`. And yep, I wrote current checks with `SKIP` being sorted out ASAP, e.g.:

```python
    def lint_manual_test_path_exists(self) -> LinterReturn:
        """ T007: manual test path is not an actual path """

        if not self.manual:
            yield LinterOutcome.SKIP, 'not a manual test'
            return

        if not self.path:
            yield LinterOutcome.FAIL, 'directory path is not set'
            return

        if not self.manual_test_path.exists():
            yield LinterOutcome.FAIL, f'manual test path "{self.manual_test_path}" does not exist'
            return

        yield LinterOutcome.PASS, f'manual test path "{self.manual_test_path}" does exist'
```

So it would be consistent with rest of the checks.

Reply:
Updated !